R/civis_ml_utils.R

```diff
@@ -163,6 +163,26 @@ get_metric <- function(model, name = NULL) {
   }
 }
 
+
+#' Get model feature importance
+#' @param model Object from \code{\link{civis_ml}}
+#' @return A matrix of features and their importance, ranked in descending order of importance
+#' @export
+get_feature_importance <- function(model){
+
+  if (is.null(model$metrics$model$parameters$feature_importances)) stop("Feature importance data not available.")
+
+  variable_order <- order(model$metrics$model$parameters$feature_importances)
+  variable_names <- model$metrics$model$parameters$relvars[rev(variable_order)]
+  importance <- model$metrics$model$parameters$feature_importances[rev(variable_order)]
+
+  feature_importance_df <- data.frame('importance' = importance)
+  rownames(feature_importance_df) <- variable_names
+  feature_importance_df
+}
```

Comment:
I don't have a strong preference on the output format, but is there a reason to prefer

1. A dataframe with a single column, "importance" and rownames

to

2. A dataframe with two columns, "importance" and "variable_name"

?  It seems it might be slightly easier to work with the latter to me.